activesupport/lib/active_support/message_verifier.rb

```diff
@@ -106,18 +134,74 @@ def verify(signed_message)
     #
     #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
     #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
-    def generate(value)
-      data = encode(@serializer.dump(value))
-      "#{data}--#{generate_digest(data)}"
+    class Claims
```

Comment:
Let's set ourselves up right and put this into its own file the verifier and encryptor can both use.